File: app/src/main/java/com/pietrantuono/podcasts/singlepodcast/customviews/EpisodeHolder.java
package com.pietrantuono.podcasts.singlepodcast.customviews;

import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.pietrantuono.podcasts.BR;
import com.pietrantuono.podcasts.apis.PodcastEpisodeModel;
import com.pietrantuono.podcasts.singlepodcast.viewmodel.PodcastEpisodeViewModel;
import com.pietrantuono.podcasts.singlepodcast.viewmodel.ResourcesProvider;

public class EpisodeHolder extends RecyclerView.ViewHolder {
    private final ViewDataBinding dataBinding;
    private final ResourcesProvider resourcesProvider;

    public EpisodeHolder(View itemView, ResourcesProvider resourcesProvider) {
        super(itemView);
        dataBinding = DataBindingUtil.bind(itemView);
        this.resourcesProvider = resourcesProvider;
    }

    public void bind(PodcastEpisodeModel podcastEpisodeModel) {
        PodcastEpisodeViewModel podcastEpisodeViewModel = new PodcastEpisodeViewModel(podcastEpisodeModel, resourcesProvider);
        dataBinding.setVariable(BR.viewModel, podcastEpisodeViewModel);
        dataBinding.executePendingBindings();
    }
}

File: app/src/main/java/com/pietrantuono/podcasts/singlepodcast/customviews/EpisodeHolder.kt
package com.pietrantuono.podcasts.addpodcast.singlepodcast.customviews

import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView
import android.view.View
import com.pietrantuono.podcasts.BR
import com.pietrantuono.podcasts.addpodcast.singlepodcast.viewmodel.EpisodeViewModel
import com.pietrantuono.podcasts.addpodcast.singlepodcast.viewmodel.ResourcesProvider
import models.pojos.Episode


class EpisodeHolder(itemView: View, private val resourcesProvider: ResourcesProvider) : RecyclerView.ViewHolder(itemView) {
    private val dataBinding: ViewDataBinding

    init {
        dataBinding = DataBindingUtil.bind<ViewDataBinding>(itemView)
    }

    fun bind(episode: Episode, clickListener: EpisodesAdapter.OnItemClickListener?) {
        val podcastEpisodeViewModel = EpisodeViewModel(episode, resourcesProvider)
        dataBinding.setVariable(BR.viewModel, podcastEpisodeViewModel)
        dataBinding.executePendingBindings()
        dataBinding.root.setOnClickListener{
            clickListener?.onItemClick(episode)
        }
    }
}
